client/my-sites/hosting/github/github-connect-card/search-branches.tsx

```diff
@@ -22,20 +22,31 @@ export const SearchBranches = ( {
 }: SearchBranchesProps ) => {
 	const { __ } = useI18n();
 
-	const { data: branches, isLoading: isLoading } = useGithubBranchesQuery(
-		siteId,
-		repoName,
-		connectionId,
-		{
-			onSuccess( branches ) {
-				if ( branches.length < 30 ) {
-					onSelect( branches[ 0 ] );
-				}
-			},
-		}
-	);
+	const { data: branches, isFetching } = useGithubBranchesQuery( siteId, repoName, connectionId, {
+		onSuccess( branches ) {
+			if ( branches.length < 30 ) {
+				onSelect( branches[ 0 ] );
+			}
+		},
+	} );
 
-	if ( branches && branches.length < 30 ) {
+	if ( ! repoName || isFetching ) {
+		return (
+			<div style={ { position: 'relative' } }>
+				<FormTextInput
+					id="branch"
+					disabled={ ! repoName || isFetching }
+					className="connect-branch__repository-field"
+					placeholder={ branches ? __( 'Type a branch' ) : __( 'Choose a branch' ) }
+					onChange={ ( e: ChangeEvent< HTMLInputElement > ) => onSelect( e.target.value ) }
+					value={ selectedBranch }
+				/>
+				{ isFetching && <Spinner className="connect-branch__loading" /> }
+			</div>
+		);
+	}
+
+	if ( branches && branches.length > 0 && branches.length < 30 ) {
```

Comment:
Not related directly to this PR: are we not supporting more than 30 branches, then? 😄

Reply:
Showing more than 30 branches on a select field would be not very good 😄 

The branches are sorted by updated date and the default branch is always on top. If there are more than 30 branches we're forcing the user to type its name because GH API has no search functionality for branches, unfortunately.

Reply:
@mpkelly could move that 30 into a constant in this PR.